src/registrar/assets/js/get-gov-admin.js

```diff
@@ -750,3 +760,107 @@ function initializeWidgetOnList(list, parentId) {
+        let organizationType = document.querySelector("#id_organization_type");
+        if ($federalAgency && organizationType) {
+            // Execute this function once on load
+            handleFederalAgencyChange($federalAgency, organizationType);
+
+            // Attach the change event listener
+            $federalAgency.on("change", function() {
+                handleFederalAgencyChange($federalAgency, organizationType);
+            });
+        }
+        
+        // Handle dynamically hiding the urbanization field
+        let urbanizationField = document.querySelector(".field-urbanization");
+        let stateTerritory = document.querySelector("#id_state_territory");
+        if (urbanizationField && stateTerritory) {
+            // Execute this function once on load
+            handleStateTerritoryChange(stateTerritory, urbanizationField);
+
+            // Attach the change event listener for state/territory
+            stateTerritory.addEventListener("change", function() {
+                handleStateTerritoryChange(stateTerritory, urbanizationField);
+            });
+        }
+    });
+
+    function handleFederalAgencyChange(federalAgency, organizationType) {
+        // Set the org type to federal if an agency is selected
+        let selectedText = federalAgency.find("option:selected").text();
+        if (selectedText !== "Non-Federal Agency" && selectedText) {
+            if (organizationType.value !== "federal") {
+                organizationType.value = "federal";
+            }
+            // Set the SO field
+        }else if (selectedText === "Non-Federal Agency" && organizationType.value === "federal") {
+            organizationType.value = "";
+            // Set the SO field
+        }
+        
+        // There isn't a senior official associated with null records and non federal agencies
+        if (!selectedText || selectedText === "Non-Federal Agency") {
+            return;
+        }
+
+        // Get the associated senior official with this federal agency
+        let $seniorOfficial = django.jQuery("#id_senior_official");
+        if (!$seniorOfficial) {
+            console.log("Could not find the senior official field");
+            return;
+        }
+
+        let seniorOfficialApi = document.querySelector("#senior_official_from_agency_json_url").value;
+        fetch(`${seniorOfficialApi}?agency_name=${selectedText}`)
+        .then(response => response.json())
+        .then(data => {
+            if (data.error) {
+                console.error('Error in AJAX call: ' + data.error);
+                return;
+            }
+
+            let seniorOfficialId = data.id;
+            let seniorOfficialName = [data.first_name, data.last_name].join(" ");
+            if (!seniorOfficialId || !seniorOfficialName || !seniorOfficialName.trim()){
+                console.error("Could not assign current Senior Official: no values found.")
```

Comment:
This is really great stuff! One question: would it be better to empty the SO field if a federal agency is selected that doesn't have an SO that is in the CSV? Right now when you switch from an agency that does have an SO to one that doesn't, it keeps the previous. If that's the behavior we want then that's perfect, just want to make sure it's a conscious choice.

Reply:
Yes, agreed it should clear the SO selection if the agency field changes.